src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -220,6 +203,19 @@ protected SyntaxNode GetCallSiteContainerFromOutermostMoveInVariable(Cancellatio
                 return declStatement.Parent;
             }
 
+            protected sealed override SyntaxNode GetOutermostCallSiteContainerToProcess(CancellationToken cancellationToken)
+            {
+                var callSiteContainer = GetCallSiteContainerFromOutermostMoveInVariable(cancellationToken);
+                if (callSiteContainer != null)
+                {
+                    return callSiteContainer;
+                }
+                else
+                {
+                    return this.SelectionResult.GetOutermostCallSiteContainerToProcess(cancellationToken);
+                }
+            }
```

Comment:
this was logic from all 3 subtypes now moved to the base type.  They all share the top half in common.  The bottom half now moves to the selectionResult type which answers that question.